mods/ra/maps/ant-03/ant-03-AI.lua

```diff
@@ -0,0 +1,104 @@
+--[[
+   Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+NorthNestPath = { NorthPath1, NorthPath2 }
+SouthNestPath = { SouthPath1, SouthPath2 }
+BaseNestPath = { NorthPath1, SouthPath1 }
+
+AntSquad =
+{
+	easy = { "warriorant", "warriorant", "warriorant" },
+	normal = { "warriorant", "warriorant", "warriorant", "warriorant" },
+	hard = { "warriorant", "warriorant", "warriorant", "warriorant", "warriorant" }
+}
+
+ActivateHive1 = function()
+	local ants = Reinforcements.Reinforce(USSR, AntSquad, { Hive1.Location })
+	Utils.Do(ants, IdleHunt)
+
+	Trigger.AfterDelay(DateTime.Minutes(1), ActivateHive1)
+end
+
+ActivateHive2 = function()
+	local ants = Reinforcements.Reinforce(USSR, AntSquad, { Hive2.Location })
+	local path = Utils.Random(NorthNestPath)
+	Utils.Do(ants, function(ant)
+		ant.AttackMove(path.Location)
+		IdleHunt(ant)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(1), ActivateHive2)
+end
+
+ActivateHive3 = function()
+	local ants = Reinforcements.Reinforce(USSR, AntSquad, { Hive3.Location })
+	local path = Utils.Random(NorthNestPath)
+	Utils.Do(ants, function(ant)
+		ant.AttackMove(path.Location)
+		IdleHunt(ant)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(1), ActivateHive3)
+end
+
+ActivateHive4 = function()
+	local ants = Reinforcements.Reinforce(USSR, AntSquad, { Hive4.Location })
+	local path = Utils.Random(SouthNestPath)
+	Utils.Do(ants, function(ant)
+		ant.AttackMove(path.Location)
+		IdleHunt(ant)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(1), ActivateHive4)
+end
+
+ActivateHive5 = function()
+	local ants = Reinforcements.Reinforce(USSR, AntSquad, { Hive5.Location })
+	local path = Utils.Random(SouthNestPath)
+	Utils.Do(ants, function(ant)
+		ant.AttackMove(path.Location)
+		IdleHunt(ant)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(1), ActivateHive5)
+end
+
+ActivateHive6 = function()
+	local ants = Reinforcements.Reinforce(USSR, AntSquad, { Hive6.Location })
+	local path = Utils.Random(BaseNestPath)
+	Utils.Do(ants, function(ant)
+		ant.AttackMove(path.Location)
+		IdleHunt(ant)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(1), ActivateHive6)
+end
+
+ActivateHive7 = function()
+	local ants = Reinforcements.Reinforce(USSR, AntSquad, { Hive7.Location })
+	local path = Utils.Random(SouthNestPath)
+	Utils.Do(ants, function(ant)
+		ant.AttackMove(path.Location)
+		IdleHunt(ant)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(1), ActivateHive7)
+end
+
+ActivateAntHives = function()
+	local difficulty = Map.LobbyOption("difficulty")
+	AntSquad = AntSquad[difficulty]
```

Comment:
Is it normal to reuse global variables like this?  It would seem more natural to have two separate variables for the squad for this game and the table of all possible squads.

Reply:
It's consistent with what we do with difficulty levels in general (check Allies08 for example), but I'm not sure that's the cleanest way overall.